control_software/src/paper_gpu/file_conversion.py

```diff
@@ -230,8 +224,10 @@ def make_uvh5_file(filename, metadata_file, data_file):
         to_enu=True,
     )
 
-    # build frequency information
-    bandwidth = 250e6
+    # build frequency information from redis
+    rd = redis.Redis("redishost", decode_responses=True)
+    sample_freq = float(rd["feng:sample_freq"])
+    bandwidth = sample_freq / 2.0
     nchans = int(2048 // 4 * 3)  # number of channels we're writing
     nchans_f = 8192  # total number of channels in SNAP
```

Comment:
Sorry, couple last things here:
1) nchans_f should be rd["feng:samples_per_mcnt"] // 2
2) I don't understand the math of nchans. I think it should derive from nchans_f as: nchans_f // nchan_sum * 3 // 4, which is numerically equivalent, but robust against nchans_f changes.
3) similarly, start_chan should be derived as nchans_f // 16 * 3